### PR TITLE
Update generator for roster slots and create one for in season draft picks

### LINKED ISSUE #275: Hide "Create Roster Slots" when not needed
Hide "Create Roster Slots" in Championship show when they are already present to it is not an event.

### PR DESCRIPTION
**Pull request by axelclark**
Limit when create roster slot button is shown
* Only shown to admin, category: event, and when none are present
* Reduces risk of admin accidently creating unnecessary slots
* Closes #275

### CODE REVIEW

File: web/models/roster_position.ex [hunk 1]
```diff
@@ -60,6 +60,25 @@ defmodule Ex338.RosterPosition do
...
+      where: r.status == "active"
+  end
+
+  def all_draft_picks(query) do
+    from r in query,
+      join: p in assoc(r, :fantasy_player),
+      where: p.draft_pick == true
+  end
+
+
```
Comment (axelclark) [inline web/models/roster_position.ex]: There should be no more than 1 consecutive blank lines.

**Comment by axelclark** [comment]
Add controller to create in season draft picks

* Enables admin to create in season draft picks from roster positions
* Creates a draft pick for each position with a draft pick player
* Only displays button to admin when championship has in season draft and no picks yet
* Closes #265

File: web/models/in_season_draft_pick.ex [hunk 2]
```diff
@@ -22,10 +24,30 @@ defmodule Ex338.InSeasonDraftPick do
...
+    data: %{draft_pick_asset: %{fantasy_team: %{fantasy_league_id: league_id}}}}
+    = changeset) do
+
+    num_picks = 1
+    [next_pick] = InSeasonDraftPick.Store.next_picks(league_id, num_picks)
+
+    compare_to_next_pick(changeset.data.id, next_pick.id, changeset)
+  end
+
+  defp validate_is_next_pick(changeset), do: changeset
```
Comment (credoci) [inline web/models/in_season_draft_pick.ex]: Parameter `changeset` has same name as a function in the same module.

File: web/models/in_season_draft_pick.ex [hunk 3]
```diff
@@ -22,10 +24,30 @@ defmodule Ex338.InSeasonDraftPick do
...
+
+  defp compare_to_next_pick(next_pick, next_pick, changeset), do: changeset
+
+  defp compare_to_next_pick(_pick, _next_pick, changeset) do
```
Comment (credoci) [inline web/models/in_season_draft_pick.ex]: Parameter `changeset` has same name as a function in the same module.

File: web/models/in_season_draft_pick.ex [hunk 4]
```diff
@@ -22,10 +24,30 @@ defmodule Ex338.InSeasonDraftPick do
...
   def owner_changeset(pick, params \\ %{}) do
     pick
     |> cast(params, [:drafted_player_id])
     |> validate_required([:drafted_player_id])
+    |> validate_is_next_pick
+  end
+
+  defp validate_is_next_pick(%{
+    data: %{draft_pick_asset: %{fantasy_team: %{fantasy_league_id: league_id}}}}
+    = changeset) do
```
Comment (credoci) [inline web/models/in_season_draft_pick.ex]: Parameter `changeset` has same name as a function in the same module.

File: web/models/in_season_draft_pick.ex [hunk 5]
```diff
@@ -22,10 +24,30 @@ defmodule Ex338.InSeasonDraftPick do
...
+
+    num_picks = 1
+    [next_pick] = InSeasonDraftPick.Store.next_picks(league_id, num_picks)
+
+    compare_to_next_pick(changeset.data.id, next_pick.id, changeset)
+  end
+
+  defp validate_is_next_pick(changeset), do: changeset
+
+  defp compare_to_next_pick(next_pick, next_pick, changeset), do: changeset
```
Comment (credoci) [inline web/models/in_season_draft_pick.ex]: Parameter `changeset` has same name as a function in the same module.

File: web/models/in_season_draft_pick.ex [hunk 6]
```diff
@@ -22,10 +24,30 @@ defmodule Ex338.InSeasonDraftPick do
     |> validate_required([:position, :draft_pick_asset_id, :championship_id])
   end
 
+
```
Comment (axelclark) [inline web/models/in_season_draft_pick.ex]: There should be no more than 1 consecutive blank lines.

**Comment by axelclark** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/axelclark/ex338/pulls/276.

**Pull request merged**